app.rb

```diff
@@ -0,0 +1,167 @@
+  end
+
+  def check(options)
+    case options
+    when 1
+      list_books
+    when 2
+      list_people
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    end
+  end
+
+  def execute
+    choice = 0
+    while choice != 7
+      menu
+      puts
+      puts
+      print '[Enter 1-7]: '
+      choice = gets.chomp.strip.to_i
+      check(choice)
+      puts
+      puts
+    end
+  end
+
+  def create_book
+    puts
+    print 'Title: '
+    title = gets.chomp.strip.capitalize
+    print 'Author: '
+    author = gets.chomp.strip.capitalize
+    @books.push(Book.new(title, author))
+  end
+
+  def list_books
+    puts
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def create_person
+    puts
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice = gets.chomp.strip.to_i
+    case choice
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+  end
+
+  def create_student
+    puts
+    print 'Age: '
+    age = gets.chomp.strip.to_i
+    while age <= 0 || age >= 100
+      print 'Please input a valid age: (1 - 100): '
+      age = gets.chomp.strip.to_i
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.strip.upcase
+    case permission
+    when 'Y'
+      permission = true
+    when 'N'
+      permission = false
+    end
+    @people << Student.new(age, nil, name, parent_permission: permission)
+    puts 'Person created successfully'
+  end
+
+  def create_teacher
+    puts
+    print 'Age: '
+    age = gets.chomp.strip.to_i
+    while age <= 0 || age >= 100
+      print 'Please input a valid age (1 - 100): '
+      age = gets.chomp.strip.to_i
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Specialization: '
+    specialization = gets.chomp.strip
+    @people << Teacher.new(age, specialization, name)
+    puts 'Person created successfully'
+  end
+
+  def create_rental
+    puts
+    puts 'Select a book from the following list by number'
+    list_books
+    book_choice = gets.chomp.to_i
+    while book_choice.negative? || book_choice >= @books.length
+      print "Please enter a number within 0 - #{@books.length - 1} range: "
+      book_choice = gets.chomp.to_i
+    end
+    book = @books[book_choice]
+    puts 'Select a person from the following list by number (not id)'
+    list_people
+    person_choice = gets.chomp.to_i
+    while person_choice.negative? || person_choice >= @people.length
+      print "Please enter a number within 0 - #{@people.length - 1} range: "
+      person_choice = gets.chomp.to_i
+    end
+    person = @people[person_choice]
+    print 'Enter date of booking: (yyyy/mm/dd) : '
+    date = gets.chomp.strip
+    person.add_rental(date, book)
```

Comment:
- [ ] When I'm trying to create a rental, I get the following error because on line 141 you are calling method `add_rental` from person class that does not exist on that class. There's typo please change `add_rental` with `add_rentals` to fix this issue.

![image](https://user-images.githubusercontent.com/83803414/186909044-9bd306a3-bd31-4d9a-8476-d20ffdd89f9f.png)

Reply:
Thankyou 😊